src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -121,12 +121,19 @@ function useChatUtilsHook(): UseChatUtilsHook {
     });
   };
 
-  const goToChatScreen = (channel: ChannelList, from) => {
+  const goToChatScreen = (channel: ChannelList, from: AllowedGoToChatScreen) => {
     setChat({
       ...chat,
       selectedChannel: channel
     });
     setChannelAsRead(channel);
+    if (from === 'CONTACT_SCREEN') {
+      return openChat(
+        'SignedChatScreen',
+        channel?.channelType === ANON_PM ? 'AnonymousChannelList' : 'ChannelList'
+      );
+    }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in the `goToChatScreen` function seems to be a bit confusing. If `from === 'CONTACT_SCREEN'`, it will return and not execute the following code, which might be necessary for other cases as well. Consider revising this logic to ensure that all necessary steps are executed for each case.
```diff
-    if (from === 'CONTACT_SCREEN') {
-      return openChat(
-        'SignedChatScreen',
-        channel?.channelType === ANON_PM ? 'AnonymousChannelList' : 'ChannelList'
-      );
-    }
-
-    if (channel?.channelType === ANON_PM) {
-      if (from === GROUP_INFO) {
-        return openChat('AnonymousChatScreen', 'AnonymousChannelList');
-      }
+    let screenName = 'SignedChatScreen';
+    let listName = channel?.channelType === ANON_PM ? 'AnonymousChannelList' : 'ChannelList';
+
+    if (channel?.channelType === ANON_PM && from === GROUP_INFO) {
+      screenName = 'AnonymousChatScreen';
+    }
+
+    return openChat(screenName, listName);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->